src/codec.rs

```diff
@@ -40,27 +39,28 @@ impl ToAvro for Codec {
 }
 
 impl FromStr for Codec {
-    type Err = DecodeError;
+    type Err = AvroError;
 
     fn from_str(s: &str) -> Result<Self, Self::Err> {
         match s {
             "null" => Ok(Codec::Null),
             "deflate" => Ok(Codec::Deflate),
             #[cfg(feature = "snappy")]
             "snappy" => Ok(Codec::Snappy),
-            _ => Err(DecodeError::new("unrecognized codec")),
+            _ => Err(AvroError::DecodeError("unrecognized codec".to_string())),
```

Comment:
I am happy to hear that you'd like to contribute to writing error variants. I honestly don't know how many of them is too many or too few, so a second opinion would definitely be appreciated.

I am happy to merge this one with a few more of the suggested changes so that you can do a second pass or to merge your PR into my fork. Feel free to choose what works best for you.